### PR TITLE
python3Packages.{websockets, uvicorn,python-engineio}: bump an…

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
 #68361
 
bump a few related packages, they were all broken before. wanted `nix-review` to pass.

enabled tests on uvicorn and websockets to make sure I wasn't breaking anything by disregarding version bounds.
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

```
https://github.com/NixOS/nixpkgs/pull/68939
12 package were build:
datasette gdbgui grab-site python37Packages.fastapi python37Packages.flask-socketio python37Packages.hbmqtt python37Packages.python-engineio python37Packages.python-socketio python37Packages.sanic python37Packages.starlette python37Packages.uvicorn python37Packages.websockets
```

### CODE REVIEW

File: pkgs/development/python-modules/pgadmin4/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonPackage
```
Comment (B4dM4n) [inline pkgs/development/python-modules/pgadmin4/default.nix]: This file is not used anywhere (and seems to be in the wrong folder).

File: pkgs/development/python-modules/uvicorn/default.nix [hunk 2]
```diff
@@ -29,11 +33,18 @@ buildPythonPackage rec {
...
   ];
 
-  checkPhase = ''
-    $out/bin/uvicorn --help
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "h11==0.8.*" "h11"
   '';
 
-  patches = [ ./setup.patch ];
```
Comment (B4dM4n) [inline pkgs/development/python-modules/uvicorn/default.nix]: You should also remove the `setup.patch` file.

File: pkgs/development/python-modules/uvicorn/default.nix [hunk 3]
```diff
@@ -29,11 +33,18 @@ buildPythonPackage rec {
...
+  checkInputs = [ pytest requests ];
+  checkPhase = ''
+    pytest
+  '';
+  #checkPhase = ''
+  #  $out/bin/uvicorn --help
+  #'';
```
Comment (B4dM4n) [inline pkgs/development/python-modules/uvicorn/default.nix]: With actual tests added, the old `checkPhase` should be removed.

**Review by B4dM4n — COMMENTED** [review]
The updated versions seem fine, but I put some comments inline.

I'm also not a fan of reordering lines, just for the sake of them being alphabetically sorted (like in `python-engineio/default.nix`). It makes it harder to reason about the actual changes. But this is just my personal opinion.

File: pkgs/development/python-modules/pgadmin4/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/pgadmin4/default.nix]: oops, i didn't mean to include this. xD

Comment (jonringer) [reply pkgs/development/python-modules/pgadmin4/default.nix]: accidently included while trying to do git rebase -i

File: pkgs/development/python-modules/uvicorn/default.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/development/python-modules/uvicorn/default.nix]: Yea, looks like i messed up a git rebase -i :(

File: pkgs/development/python-modules/uvicorn/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/python-modules/uvicorn/default.nix]: done

File: pkgs/development/python-modules/pgadmin4/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/pgadmin4/default.nix]: removed, sorry about that. `git add .` is dangerous. lol

File: pkgs/development/python-modules/uvicorn/default.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/development/python-modules/uvicorn/default.nix]: removed

**Comment by wd15** [comment]
Agree regarding reordering, but leave it to your discretion.

Also, is Nixpkgs moving to `fetchFromGitHub` rather than `fetchPypi` in general?

Result of `nix-review` [1](https://github.com/Mic92/nix-review) on Ubuntu using `x86_64`.
<details>
  <summary>12 packages were built:</summary>

  - datasette (python37Packages.datasette)
  - gdbgui (python37Packages.datasette)
  - grab-site (python37Packages.datasette)
  - python37Packages.fastapi (python37Packages.datasette)
  - python37Packages.flask-socketio (python37Packages.datasette)
  - python37Packages.hbmqtt (python37Packages.datasette)
  - python37Packages.python-engineio (python37Packages.datasette)
  - python37Packages.python-socketio (python37Packages.datasette)
  - python37Packages.sanic (python37Packages.datasette)
  - python37Packages.starlette (python37Packages.datasette)
  - python37Packages.uvicorn (python37Packages.datasette)
  - python37Packages.websockets (python37Packages.datasette)
</details>

**Review by wd15 — APPROVED** [review]

**Comment by jonringer** [comment]
@wd15 if the pypi package doesn't include tests, then I go to fetchFromGitHub route, as that definitely will have tests (if the they have tests at all)

**Comment by jonringer** [comment]
@B4dM4n I was verifying that the expression had the needed dependencies and only the needed dependencies, having them in ad-hoc order wasn't fun. plus in vim, it's just a `:!sort` command away

**Comment by Mic92** [comment]
[detached HEAD d3f56ac32a5] python3Packages.websockets: 7.0 -> 8.0.2
 Author: Jonathan Ringer <jonringer117@gmail.com>
 Date: Mon Sep 16 13:24:14 2019 -0700
 1 file changed, 11 insertions(+), 9 deletions(-)
Press Enter to continue
[detached HEAD 91abf952f2e] python3Packages.uvicorn: 0.8.4 -> 0.9.0
 Author: Jonathan Ringer <jonringer117@gmail.com>
 Date: Mon Sep 16 15:07:28 2019 -0700
 2 files changed, 16 insertions(+), 21 deletions(-)
 delete mode 100644 pkgs/development/python-modules/uvicorn/setup.patch
Press Enter to continue
[detached HEAD 6825f045df3] python3Packages.python-engineio: 3.4.3 -> 3.9.3
 Author: Jonathan Ringer <jonringer117@gmail.com>
 Date: Mon Sep 16 13:29:34 2019 -0700
 1 file changed, 12 insertions(+), 14 deletions(-)

**Pull request merged**